R/plotFeatureLoadings.R

```diff
@@ -0,0 +1,404 @@
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- logNormCounts(tse)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", prevalence=0.99, update.tree = TRUE)
+#' tse <- runPCA(tse, name = "PCA", ncomponents = 2)
+#' plotFeatureLoadings(loadings_matrix, method= "PCA", plot = "Screeplot")
+NULL
+
+#' @rdname plotFeatureLoadings
+setGeneric("plotFeatureLoadings", signature = c("object"),
+           function(object, ...)
+             standardGeneric("plotFeatureLoadings"))
+
+
+#' @rdname plotFeatureLoadings
+#' @export 
+setMethod("plotFeatureLoadings", signature = c(object = "TreeSummarizedExperiment"),
+          function(object,
+                   method = "PCA", plot = "Heatmap",
+                   ...){
+            
+            .check_parameters(object,
+                              method = method, plot = plot, ...)   
+            
+            if (method == "PCA") {
+              loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+              df <- data.frame(loadings_matrix)
+              df[["Feature"]] <- rownames(loadings_matrix)
+              df[["Feature"]] <- factor(df[["Feature"]], levels = df[["Feature"]])
+              df <- df[ order(abs(df[["PC1"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC1"]]), ]
+              df <- df[ order(abs(df[["PC2"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC2"]]), ]
+            } else {
+              stop("These methods are yet to be implemented", call. = FALSE)
+            }
+            
+            if (plot == "Tree") {
+              
+              phylo <- rowTree(object)
+              circ <- ggtree(phylo, layout = "circular")
+              df <- rowData(object)
+              color <- randomcoloR::distinctColorPalette(
+                length(
+                  unique(
+                    df$Phylum
+                  )
+                )
+              )
+              df <- data.frame(Class = df$Phylum)
+              rownames(df) <- phylo$tip.label
+              
+              
+              df2 <- data.frame(abs(loadings_matrix))
+              rownames(df2) <- phylo$tip.label
+              
+              
+              p <- gheatmap(
+                p = circ,
+                data = df,
+                offset = -.1,
+                width = .1,
+                colnames_angle = 95,
+                colnames_offset_y = .5,
+                font.size = 4,
+                color = "black") +
+                ggplot2::scale_fill_manual(
+                  values = color,
+                  name = "Class"
+                )
+              
+              for(i in 1:2){
+                if(i == 1){
+                  p <- p +
+                    ggnewscale::new_scale_fill()
+                }
+                df3 <- dplyr::select(
+                  df2, (all_of(i))
+                )
+                
+                p <- gheatmap(
```

Comment:
INstead of using new dependency, you could check https://github.com/microbiome/miaViz/blob/dde325321015e657ff2cf904c9feb8387c9eb2f4/R/plotPrevalence.R#L576

Reply:
For tree plotting, we have this function https://github.com/microbiome/miaViz/blob/dde325321015e657ff2cf904c9feb8387c9eb2f4/R/plotTree.R#L1031 Try to check if you can use that code.

For maintenance, it is always easier if we can use and improve the existing code rather than having multiple implementations on the same thing (tree plotting might not be relevant here though, just a general note. However, for heatmap the existing code might work)